src/types/binary/word.ts

```diff
@@ -208,4 +210,58 @@ export class Word {
     const hexString = this.value.toString(16)
     return hexString.padStart(8, '0')
   }
+
+  /**
+   * throws exception if offset is not smaller than 32
+   * @param bitOffset 0-indexed offset
+   */
+  private checkIfBitOffsetInWordRange(bitOffset: number): void {
+    if (bitOffset >= Word.NUMBER_OF_BITS || bitOffset < 0) {
+      throw new VirtualBoardError(
+        'Offset is not within Word range',
+        VirtualBoardErrorType.BitOutOfTypeRange
+      )
+    }
+  }
+
+  /**
+   *
+   * @param bitOffset
+   * @returns
+   */
+  public isBitSet(bitOffset: number): boolean {
```

Comment:
I saw that you added these functions `isBitSet`, `setBit` and `clearBit` to the word type. I actually did almost the same for the `Byte` type for the dip switches because there I also need the possibility to set bits. But the code is basically the same... Should I just duplicate them in the `Byte` type or do we maybe do something like a super class or move this to utils.ts?

Reply:
I just noticed that the functions like `toUnsignedInteger` etc are also kind of the same for the three types so I guess we should stick to that and duplicate the function again for `Byte` and maybe later `HalfWord`...

Reply:
As you already noticed, there is already some similar/duplicate code across the binary types. We may think about creating a subtype for those in the future.

I would vote against the usage of a `utils.ts` module in that case, and instead we might create a base class (or just duplicate the code for now).